### PR TITLE
Solution for `React Tabs`

### PR DESCRIPTION
**Pull request by jstmpelowycz**
**[DEMO LINK](https://jstmpelowycz.github.io/react_tabs/)**

### CODE REVIEW

**Review by aspurgilus — APPROVED** [review]
Nice solution using useMemo & useCallback everywhere. May be someday I could do it too.

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+
+export const App: FC = memo(() => {
+  const [selectedTab, setTab] = useState(tabs[0]);
 
-const tabs: Tab[] = [
-  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-];
+  const handleTabSelected = useCallback(setTab, []);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:11]: redundant

File: src/components/Tab/Tab.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+} from 'react';
+
+interface Props {
+  tab: Tab;
+  selectedTabId: string;
+  onTabSelected: (tab: Tab) => void;
+}
+
+export const Tab: FC<Props> = memo(
+  ({ tab, selectedTabId, onTabSelected }) => {
```
Comment (Ihor-Karpyn) [inline src/components/Tab/Tab.tsx:12]:
```suggestion
export const Tab: FC<Props> = memo(({
  tab, selectedTabId, onTabSelected
}) => {
```

File: src/components/Tab/Tab.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const { id, title, content } = useMemo(() => tab, []);
```
Comment (Ihor-Karpyn) [inline src/components/Tab/Tab.tsx:13]: redundant
```suggestion
    const { id, title, content } = tab;
```

File: src/components/Tab/Tab.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const isSelected = useMemo(
+      () => selectedTabId === id,
+      [selectedTabId],
+    );
+
+    const handleButtonClick = useCallback(() => {
+      if (!isSelected) {
+        onTabSelected(tab);
+      }
+    }, [selectedTabId]);
```
Comment (Ihor-Karpyn) [inline src/components/Tab/Tab.tsx:24]: if you don't pass callbacl to other component, `useCallback `redudant

File: src/components/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { Tab } from '../Tab';
+
+interface Props {
+  tabs: Tab[];
+  selectedTabId: string;
+  onTabSelected: (tab: Tab) => void;
+}
+
+export const Tabs: FC<Props> = memo(
+  ({ tabs, selectedTabId, onTabSelected }) => {
```
Comment (Ihor-Karpyn) [inline src/components/Tabs/Tabs.tsx:11]:
```suggestion
export const Tabs: FC<Props> = memo(({
  tabs, selectedTabId, onTabSelected
}) => {
```

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**